.github/workflows/scheduled-event-processor.yml

```diff
@@ -20,6 +20,8 @@ permissions: {}
 
 jobs:
   event-handler:
+    # prevents the job from being executed on forks
+    if: github.repository == 'azure/azure-sdk-for-net'
```

Comment:
```suggestion
    if: startsWith(github.repository, 'azure') != true
```

Reply:
I think the job level secret check wouldn't work due to [this restriction,](https://github.com/orgs/community/discussions/26726#discussioncomment-3253118) but if you like, I can flip this PR to include the secrets check on every step